File: sdks/java/src/main/java/dev/atap/sdk/model/Revocation.java
package dev.atap.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * A revocation entry for a previously-granted approval.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class Revocation {

    @JsonProperty("id")
    private String id;

    @JsonProperty("approval_id")
    private String approvalId;

    @JsonProperty("approver_did")
    private String approverDid;

    @JsonProperty("revoked_at")
    private String revokedAt;

    @JsonProperty("expires_at")
    private String expiresAt;

    public Revocation() {
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getApprovalId() { return approvalId; }
    public void setApprovalId(String approvalId) { this.approvalId = approvalId; }

    public String getApproverDid() { return approverDid; }
    public void setApproverDid(String approverDid) { this.approverDid = approverDid; }

    public String getRevokedAt() { return revokedAt; }
    public void setRevokedAt(String revokedAt) { this.revokedAt = revokedAt; }

    public String getExpiresAt() { return expiresAt; }
    public void setExpiresAt(String expiresAt) { this.expiresAt = expiresAt; }
}

File: sdks/php/src/Model/Revocation.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Model;

/**
 * A revocation entry for a previously-granted approval.
 */
final class Revocation
{
    public function __construct(
        public readonly string $id,
        public readonly string $approvalId,
        public readonly string $approverDid = '',
        public readonly string $revokedAt = '',
        public readonly string $expiresAt = '',
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            approvalId: (string) ($data['approval_id'] ?? ''),
            approverDid: (string) ($data['approver_did'] ?? ''),
            revokedAt: (string) ($data['revoked_at'] ?? ''),
            expiresAt: (string) ($data['expires_at'] ?? ''),
        );
    }
}
